packages/tools/src/tools/SculptorTool.ts

```diff
@@ -0,0 +1,590 @@
+      return;
+    }
+
+    this.commonData.activeAnnotationUID = closestAnnotationUID;
+  }
+
+  /**
+   * Finds the nearest handle to the mouse cursor for all freehand
+   * data on the element.
+   *
+   * @param eventData - Data object associated with the event.
+   */
+  private getClosestFreehandToolOnElement(eventData: any): string {
+    const { element } = eventData;
+    const enabledElement = getEnabledElement(element);
+    const { viewport } = enabledElement;
+    const config = this.configuration;
+
+    const annotations = this.filterSculptableAnnotationsForElement(element);
+
+    if (!annotations?.length) {
+      return;
+    }
+
+    const canvasPoints = eventData.currentPoints.canvas;
+
+    const closest = {
+      distance: Infinity,
+      toolIndex: undefined,
+      annotationUID: undefined,
+    };
+
+    for (let i = 0; i < annotations?.length; i++) {
+      if (annotations[i].isLocked || !annotations[i].isVisible) {
+        continue;
+      }
+
+      const distanceFromTool = distanceFromPoint(
+        viewport,
+        annotations[i].data,
+        canvasPoints
+      );
+
+      if (distanceFromTool === -1) {
+        continue;
+      }
+
+      if (distanceFromTool < closest.distance) {
+        closest.distance = distanceFromTool;
+        closest.toolIndex = i;
+        closest.annotationUID = annotations[i].annotationUID;
+      }
+    }
+
+    this.commonData.isEditingOpenContour =
+      !annotations[closest.toolIndex].data.contour.closed;
+
+    config.referencedToolName =
+      annotations[closest.toolIndex].metadata.toolName;
+
+    return closest.annotationUID;
+  }
+
+  /**
+   * Event handler for MOUSE_UP during the active loop.
+   *
+   * @param evt - The event
+   */
+  private endCallback = (
+    evt: EventTypes.MouseUpEventType | EventTypes.MouseClickEventType
+  ): void => {
+    const eventData = evt.detail;
+    const { element } = eventData;
+    const config = this.configuration;
+    const enabledElement = getEnabledElement(element);
+
+    this.isActive = false;
+    this.deactivateSculpt(element);
+    resetElementCursor(element);
+
+    const { renderingEngineId, viewportId } = enabledElement;
+
+    const toolGroup = ToolGroupManager.getToolGroupForViewport(
+      viewportId,
+      renderingEngineId
+    );
+
+    const toolInstance = toolGroup.getToolInstance(config.referencedToolName);
+
+    const annotations = this.filterSculptableAnnotationsForElement(element);
+
+    const activeAnnotation = annotations.find(
+      (annotation) =>
+        annotation.annotationUID === this.commonData.activeAnnotationUID
+    );
+
+    if (toolInstance.configuration.calculateStats) {
+      activeAnnotation.invalidated = true;
+    }
+
+    triggerAnnotationModified(activeAnnotation, element);
+  };
+
+  /**
+   * Event handler for MOUSE_DRAG during the active loop.
+   *
+   * @param evt - The event
+   */
+  private dragCallback(evt: EventTypes.InteractionEventType): void {
```

Comment:
Bind this with
private dragCallback: (evt: EventTypes.InteractionEventType) => {
....

That avoids needing to re-assign the value in the constructor.